nixos/modules/services/networking/netbird.nix

```diff
@@ -1,60 +1,169 @@
-{ config, lib, pkgs, ... }:
-
-with lib;
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
 
 let
-  cfg = config.services.netbird;
+  inherit (lib)
+    attrNames
+    getExe
+    literalExpression
+    maintainers
+    mapAttrs'
+    mkDefault
+    mkEnableOption
+    mkIf
+    mkMerge
+    mkOption
+    mkPackageOption
+    nameValuePair
+    optional
+    versionOlder
+    ;
+
+  inherit (lib.types)
+    attrsOf
+    port
+    str
+    submodule
+    ;
+
   kernel = config.boot.kernelPackages;
-  interfaceName = "wt0";
-in {
+
+  cfg = config.services.netbird;
+in
+{
   meta.maintainers = with maintainers; [ misuzu ];
 
+  disabledModules = [ "services/networking/netbird.nix" ];
+
   options.services.netbird = {
     enable = mkEnableOption (lib.mdDoc "Netbird daemon");
     package = mkPackageOption pkgs "netbird" { };
-  };
-
-  config = mkIf cfg.enable {
-    boot.extraModulePackages = optional (versionOlder kernel.kernel.version "5.6") kernel.wireguard;
 
-    environment.systemPackages = [ cfg.package ];
+    tunnels = mkOption {
```

Comment:
This probably needs a release note.
Also it's not clear how to actually use this option, what arguments should be passed to `netbird up`? Is `services.netbird.enabled` should be set? This kind of stuff deserves a documentation in `meta.doc`